builds/test_python_runtime.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+ARCHIVE_FILEPATH="${1:?"Error: The filepath of the Python runtime archive must be specified as the first argument."}"
+
+# We intentionally extract the Python runtime into a different directory to the one into which it
+# was originally installed before being packaged, to check that relocation works (since buildpacks
+# depend on it). Since the Python binary was built in shared mode, `LD_LIBRARY_PATH` must be set
+# when relocating, so the Python binary (which itself contains very little) can find `libpython`.
+INSTALL_DIR=$(mktemp -d)
+PYTHON="${INSTALL_DIR}/bin/python"
+export LD_LIBRARY_PATH="${INSTALL_DIR}/lib/"
+
+tar --zstd --extract --verbose --file "${ARCHIVE_FILEPATH}" --directory "${INSTALL_DIR}"
+
+# Check Python exists at the `python` alias (and not just `python3`) and can run.
+"${PYTHON}" --version
```

Comment:
[Talking out loud - eventual suggestion] If I'm reading this correctly it will expand to `"<tmpdir>/bin/python" --version` which seems to disagree with the comment which I would believe would be checking for `python -v` with that on the path? 

The mention of `python3` seems to be a red herring as it doesn't seem we're executing a named python version i.e. `python3` anywhere. 

So I guess this is REALLY checking that `"<tmpdir>/bin" has a `python` executable in it that responds to `--version` so if that bin was put on PATH it would function correctly. That checks out. 

We are exercising the python binary later in the loop. I think that what you're actually doing here is failing fast if there's a problem with either: 

- Executable permissions
- Expected file does not exist (what your comment says)
- Some problem that results in a working binary that can be executed but does not function correctly, i.e., `--version` fails.

[optional suggestion]: Perhaps we could say that (or some variation):

```suggestion
# Fail if python cannot print the version
"${PYTHON}" --version
```

Which is really already covered by the call itself, but does prevent someone from saying "why is this do-nothing line in here?" so I think it's good to leave some "here's what we're expecting to test with this code.

Reply:
No, this is checking (a) that python can successfully launch (ie it can be relocated, or that the build didn't fail catastrophically etc), (b) that we set up the `python` symlink too, here:
https://github.com/heroku/heroku-buildpack-python/blob/c233dbd97bda3ff22157561b407231d3a752818b/builds/build_python_runtime.sh#L219-L223

This is similar to the check performed during the CNB integration tests:
https://github.com/heroku/buildpacks-python/blob/de378063c83e1eec6302e79f93f189162b5bbbbf/tests/python_version_test.rs#L121-L123

I'll reword "alias" to "symlink" in the comment to make that clearer.